### ISSUE TITLE
Move each example into it's own subdirectory

### ISSUE THREAD

**Issue by dan-c-underwood** [luke5542/LanguageComparison#8]
This makes it easier for examples like Rust, Go, and D where they can benefit from using build tools such as `cargo`.

Will also make it cleaner for adding examples of different algorithms in the future.

**Comment by georgevanburgh** [comment]
+10

**Comment by luke5542** [comment]
Surely if other algorithms were given their own examples it would make sense to separate based on Algorithm rather than language...

**Comment by dan-c-underwood** [comment]
This way it's easier for people to see all examples in a particular language, and again, helps with build tools like `cargo`. Otherwise things would get messy with paths.

**Comment by georgevanburgh** [comment]
@luke5542 You'd probably want multiple levels of nesting in that case:

```
/
/sieve
/sieve/Java
/sieve/C#
...
/foobar
/foobar/Java
/foobar/C#
...
```

**Comment by luke5542** [comment]
I work on this, then, but I cannot guarantee any Docker support =P

As such, I'll likely work by maintaining the current file set and just adding things in the new directories.

**Comment by georgevanburgh** [comment]
@luke5542 If the compile script works - Docker will work.

**Comment by luke5542** [comment]
Glad to hear it. I'll try getting Docker setup eventually so I can at least verify the changes.

**Comment by luke5542** [comment]
Okay, so the initial conversion has been completed in the adequately named branch. The graph and docker setups have not been checked, though they both will need to be fixed, as I also significantly changed the root script setup to enable other tasks to be added later on.

@FireEater64 do you think you could check the Docker setup and see what needs changing?

**Comment by luke5542** [comment]
closed as of the merge of #14
